service-level-sidecar/server.js

```diff
@@ -0,0 +1,133 @@
+const crypto = require('crypto');
+const http = require('http');
+const https = require('https');
+const util = require('util');
+
+const express = require('express');
+
+const env = require('./env');
+const { getServiceReport } = require('./report');
+const log = require('./log');
+
+const app = express();
+
+const getServiceReportCb = util.callbackify(getServiceReport);
+
+/**
+ *
+ * @param {number} statusCode - HTTP status code
+ * @param {*} httpMessage - HTTP status message
+ * @param {*} httpDesc - An extra msg to include in the JSON response
+ * @returns {Error} - instance of Error
+ */
+function makeError(statusCode, httpMessage, httpDesc) {
+    const err = new Error(httpDesc || httpMessage);
+    err.status = statusCode;
+    err.httpMessage = httpMessage;
+    err.httpDesc = httpDesc;
+    return err;
+}
+
+
+// Middleware to add request logger
+app.use((req, res, next) => {
+    // eslint-disable-next-line no-param-reassign
+    req.log = log.newRequestLogger();
+    req.log.info('received request');
+    next();
+});
+
+function finishMiddleware(req, res) {
+    const info = {
+        httpCode: res.statusCode,
+        httpMessage: res.statusMessage,
+    };
+    req.log.end('finished handling request', info);
+}
+
+// Hash our api key once for reuse during the request
+// We prepend `Bearer ` to avoid having to strip it from the header value runtime for comparison
+const actualKeyHash = crypto.createHash('sha512').copy().update(`Bearer ${env.apiKey}`).digest();
```

Comment:
Why using a hash here, why not just comparing the API key directly? (we may still use `timingSafeEqual` on the compared original values)

```
`crypto.timingSafeEqual(Buffer.from(`Bearer ${env.apiKey}`), Buffer.from(req.headers.authorization))
```